Wrappers/Python/cil/optimisation/algorithms/SIRT.py

```diff
@@ -198,15 +200,17 @@ def update(self):
         self.operator.adjoint(self.r, out=self.tmp_x)
         self.x.sapyb(1.0, self.tmp_x, self._Dscaled, out=self.x)
 
-
         if self.constraint is not None:
-            # IndicatorBox allows inplace operation for proximal
-            self.constraint.proximal(self.x, tau=1, out=self.x)
+            if self._in_place: # Can be removed once in-place calculations are checked to be safe for all functions
```

Comment:
One #1664  is in, I would like this to be 
```python
try:
    self.constraint.proximal(self.x, tau=1, out=self.x)
except InPlaceError:
    self.x=self.constraint.proximal(self.x, tau=1)
```

And then the `self.in_place` flag can be removed.